Source/Core/InputCommon/ControllerInterface/ExpressionParser.cpp

```diff
@@ -427,13 +427,12 @@ class Parser
 
   ExpressionParseStatus Unary(ExpressionNode** expr_out)
   {
-    ExpressionParseStatus status;
-
     if (IsUnaryExpression(Peek().type))
     {
       Token tok = Chew();
       ExpressionNode* atom_expr;
-      if ((status = Atom(&atom_expr)) != EXPRESSION_PARSE_SUCCESS)
+      ExpressionParseStatus status = Atom(&atom_expr);
+      if (status != EXPRESSION_PARSE_SUCCESS && status != EXPRESSION_PARSE_NO_DEVICE)
```

Comment:
I'm honestly having trouble finding a good name. Could be IsExpressionParsedCorrectly but that'd just be checking if status != EXPRESSION_PARSE_SYNTAX_ERROR.

Maybe I could just simplify this to that check instead?